frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -703,6 +797,108 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
                 tags={formValues['metadata.labels']}
               />
             </div>
+            {[...fieldGroupThirdLevels].map((group) => {
+              const groupName = getGroupName(group, SpecCapability.fieldGroupThirdLevel);
+
+              const fieldList = fields
+                .filter((f) =>
+                  f.capabilities.find((c) =>
+                    c.startsWith(
+                      `${SpecCapability.fieldGroupSecondLevel}${
+                        group.split(SpecCapability.fieldGroupThirdLevel)[1]
+                      }`,
+                    ),
+                  ),
+                )
+                .filter((f) => !_.isNil(inputFor(f)));
+
+              return (
+                <div id={group} key={group}>
+                  <FieldGroup
+                    defaultExpand={_.some(
+                      fieldList,
+                      (f) => f.capabilities.includes(SpecCapability.advanced) && !f.required,
+                    )}
+                    groupName={groupName}
+                  >
+                    {[...fieldGroupSecondLevels].map((subGroup) => {
+                      const subGroupName = getGroupName(
+                        subGroup,
+                        SpecCapability.fieldGroupSecondLevel,
+                      );
+
+                      const thirdLevelFieldList = fields
+                        .filter(
+                          (f) =>
+                            f.capabilities.includes(group) && f.capabilities.includes(subGroup),
+                        )
+                        .filter((f) => !_.isNil(inputFor(f)));
+
+                      return (
+                        <div id={subGroup} key={subGroup}>
+                          <FieldGroup
+                            defaultExpand={_.some(
+                              fieldList,
+                              (f) =>
+                                f.capabilities.includes(SpecCapability.advanced) && !f.required,
+                            )}
+                            groupName={subGroupName}
+                          >
+                            {thirdLevelFieldList.map((field) => (
```

Comment:
My concern with this approach is that we're repeating similar logic for each level. To add more levels, we'd need to repeat again. Can we factor out some of the common logic?

Reply:
Agreed, this seems like it could be done with recursion.